src/axom/quest/InOutOctree.hpp

```diff
@@ -60,3362 +65,1514 @@ namespace quest
-    {
-      data[i] = vStart + i;
+          QUEST_OCTREE_DEBUG_LOG_IF(
+            DEBUG_BLOCK_1 == leafBlk || DEBUG_BLOCK_2 == leafBlk,
+            fmt::format("... intersection for triangle w/ index {} at ray "
+                        "parameter {} at point {}",
+                        tIdx,
+                        minRayParam,
+                        ray.at(minRayParam)));
+        }
+      }
     }
 
-    mesh->appendCell(data, mint::TRIANGLE);
-
-    // Log the triangle info as primal code to simplify adding a test for this case
-    if(shouldLogTris)
+    if(tIdx == MeshWrapper<DIM>::NO_CELL)
     {
-      SLIC_INFO("// Triangle " << tIdx << "\n\t"
-                               << "TriangleType tri("
-                               << "PointType::make_point" << triPos[0] << ","
-                               << "PointType::make_point" << triPos[1] << ","
-                               << "PointType::make_point" << triPos[2] << ");");
+      continue;
     }
-  }
-
-  void addOctreeBlock(DebugMesh* mesh,
-                      const BlockIndex& block,
-                      bool shouldLogBlocks) const
-  {
-    GeometricBoundingBox blockBB = m_octree.blockBoundingBox(block);
-
-    axom::IndexType vStart = mesh->getNumberOfNodes();
 
-    const SpacePt& bMin = blockBB.getMin();
-    const SpacePt& bMax = blockBB.getMax();
-
-    mesh->appendNode(bMin[0], bMin[1], bMin[2]);
-    mesh->appendNode(bMax[0], bMin[1], bMin[2]);
-    mesh->appendNode(bMax[0], bMax[1], bMin[2]);
-    mesh->appendNode(bMin[0], bMax[1], bMin[2]);
-
-    mesh->appendNode(bMin[0], bMin[1], bMax[2]);
-    mesh->appendNode(bMax[0], bMin[1], bMax[2]);
-    mesh->appendNode(bMax[0], bMax[1], bMax[2]);
-    mesh->appendNode(bMin[0], bMax[1], bMax[2]);
-
-    axom::IndexType data[8];
-    for(int i = 0; i < 8; ++i) data[i] = vStart + i;
-
-    mesh->appendCell(data, mint::HEX);
+    // Inside when the dot product of the normal with this triangle is positive
+    SpaceVector normal =
+      (tIdx == idx) ? tri.normal() : m_meshWrapper.cellPositions(tIdx).normal();
 
-    // Log bounding box info to simplify adding a test for this case
-    if(shouldLogBlocks)
-    {
-      static int counter = 0;
-      SLIC_INFO("// Block index " << block);
-      SLIC_INFO("BoundingBoxType box"
-                << ++counter << "(PointType::make_point" << bMin << ","
-                << "PointType::make_point" << bMax << ");");
-    }
+    return normal.dot(ray.direction()) > 0.;
   }
 
-private:
-  const InOutOctreeType& m_octree;
-  typename InOutOctreeType::GenerationState m_generationState;
+  SLIC_DEBUG("Could not determine inside/outside for point "
+             << queryPt << " on block " << leafBlk);
 
-  ColorsMap m_colorsMap;
-};
+  return false;  // query points on boundary might get here -- revisit this.
+}
 
 template <int DIM>
-class InOutOctreeValidator
+template <int TDIM>
+typename std::enable_if<TDIM == 2, bool>::type InOutOctree<DIM>::withinGrayBlock(
```

Comment:
There's *a lot* in this PR.

This function: `InOutOctree<2>::withinGrayBlock()` is perhaps the central change. Almost everything else here is in support of making this function work.